tensorflow/lite/kernels/lstm.cc

```diff
@@ -2071,89 +2071,85 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
-          return lstm_eval::EvalInteger8x8_8(
-              input, input_to_input_weights, input_to_forget_weights,
-              input_to_cell_weights, input_to_output_weights,
-              recurrent_to_input_weights, recurrent_to_forget_weights,
-              recurrent_to_cell_weights, recurrent_to_output_weights,
-              cell_to_input_weights, cell_to_forget_weights,
-              cell_to_output_weights, input_layer_norm_coefficients,
-              forget_layer_norm_coefficients, cell_layer_norm_coefficients,
-              output_layer_norm_coefficients, input_gate_bias, forget_gate_bias,
-              cell_gate_bias, output_gate_bias, projection_weights,
-              projection_bias, params, output_state, cell_state, output,
-              &op_data->integer_lstm_param, scratch0, scratch1, scratch2,
-              scratch3, scratch4, scratch5, scratch6, scratch7);
-          return kTfLiteOk;
-        }
       }
+      const int num_intermediate_tensors = node->intermediates->size;
+      if (num_intermediate_tensors == 5) {
+        TfLiteTensor* scratch0;
+        TF_LITE_ENSURE_OK(context,
+                          GetTemporarySafe(context, node, 0, &scratch0));
+        TfLiteTensor* scratch1;
+        TF_LITE_ENSURE_OK(context,
+                          GetTemporarySafe(context, node, 1, &scratch1));
+        TfLiteTensor* scratch2;
+        TF_LITE_ENSURE_OK(context,
+                          GetTemporarySafe(context, node, 2, &scratch2));
+        TfLiteTensor* scratch3;
+        TF_LITE_ENSURE_OK(context,
+                          GetTemporarySafe(context, node, 3, &scratch3));
+        TfLiteTensor* scratch4;
+        TF_LITE_ENSURE_OK(context,
+                          GetTemporarySafe(context, node, 4, &scratch4));
+        TfLiteTensor* scratch5;
+        TF_LITE_ENSURE_OK(context,
+                          GetTemporarySafe(context, node, 5, &scratch5));
+        return lstm_eval::EvalInteger8x8_16(
+            input, input_to_input_weights, input_to_forget_weights,
+            input_to_cell_weights, input_to_output_weights,
+            recurrent_to_input_weights, recurrent_to_forget_weights,
+            recurrent_to_cell_weights, recurrent_to_output_weights,
+            cell_to_input_weights, cell_to_forget_weights,
+            cell_to_output_weights, input_layer_norm_coefficients,
+            forget_layer_norm_coefficients, cell_layer_norm_coefficients,
+            output_layer_norm_coefficients, input_gate_bias, forget_gate_bias,
+            cell_gate_bias, output_gate_bias, projection_weights,
+            projection_bias, params, /*forward_sequence=*/true,
+            /*time_major=*/true, &op_data->integer_lstm_param, output_state,
+            cell_state, output, scratch0, scratch1, scratch2, scratch3,
+            scratch4, scratch5, CpuBackendContext::GetFromContext(context));
+      }
+      TfLiteTensor* scratch0;
```

Comment:
These line are the same with line 2077-2094. Could you move them up and remove the definition inside the if?

Reply:
Assuming you mean the definitions of scratch0 - scratch5, done.